appserver/ejb/ejb-container/src/main/java/com/sun/ejb/containers/interceptors/InterceptorUtil.java

```diff
@@ -37,120 +37,82 @@
  * only if the new code is made subject to such option by the copyright
  * holder.
  */
-// Portions Copyright [2018] [Payara Foundation and/or its affiliates]
+// Portions Copyright [2018-2019] [Payara Foundation and/or its affiliates]
 package com.sun.ejb.containers.interceptors;
 
 import java.lang.reflect.Method;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.Map;
-import java.util.HashMap;
-import java.util.HashSet;
 
-
-/**
- */
 public class InterceptorUtil {
 
-    private static Map<Class, Set<Class>> compatiblePrimitiveWrapper = new HashMap<>();
-
-     static {
-
-         Set<Class> smallerPrimitiveWrappers;
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         compatiblePrimitiveWrapper.put(byte.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Boolean.class);
-         compatiblePrimitiveWrapper.put(boolean.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Character.class);
-         compatiblePrimitiveWrapper.put(char.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         smallerPrimitiveWrappers.add(Float.class);
-         smallerPrimitiveWrappers.add(Double.class);
-         compatiblePrimitiveWrapper.put(double.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         smallerPrimitiveWrappers.add(Float.class);
-         compatiblePrimitiveWrapper.put(float.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         compatiblePrimitiveWrapper.put(int.class, smallerPrimitiveWrappers);
-
-         smallerPrimitiveWrappers = new HashSet<>();
-         smallerPrimitiveWrappers.add(Byte.class);
-         smallerPrimitiveWrappers.add(Short.class);
-         smallerPrimitiveWrappers.add(Integer.class);
-         smallerPrimitiveWrappers.add(Long.class);
-         compatiblePrimitiveWrapper.put(long.class, smallerPrimitiveWrappers);
+    private static final Map<Class<?>, Set<Class<?>>> compatiblePrimitiveWrapper = new ConcurrentHashMap<>();
```

Comment:
I am quite sure this class is used by multiple threads so `HashMap` is not adequate. Practically this does work since there is only initial writing that eventually is visible for all threads by chance but formally there are no guarantees. 

Also cleaned up the duplication and added missing generics for the file.

Reply:
HashMap only has issues with concurrent access while writing. Writing only in static initializer prevents that from happening (as the field is not accessible for reading until the static block finishes). I'd rather wrap it in `Collections.unmodifiableMap` just to make the limitation clear.

Reply:
I wasn't concerned about concurrent access but visibility across threads but as constant this should not be an issue. Will change to `unmodifiableMap`.